internal/provider/resource_integration_jira.go

```diff
@@ -0,0 +1,338 @@
+					"WIZ_INTEGRATION_JIRA_PASSWORD",
+					nil,
+				),
+			},
+		},
+		CreateContext: resourceWizIntegrationJiraCreate,
+		ReadContext:   resourceWizIntegrationJiraRead,
+		UpdateContext: resourceWizIntegrationJiraUpdate,
+		DeleteContext: resourceWizIntegrationDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func resourceWizIntegrationJiraCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	tflog.Info(ctx, "resourceWizIntegrationJiraCreate called...")
+
+	// define the graphql query
+	query := `mutation CreateIntegration($input: CreateIntegrationInput!) {
+	  createIntegration(
+	    input: $input
+	  ) {
+	    integration {
+	      id
+	    }
+	  }
+	}`
+
+	vars := &wiz.CreateIntegrationInput{}
+	vars.Name = d.Get("name").(string)
+	vars.Type = "JIRA"
+	vars.ProjectID = d.Get("project_id").(string)
+	vars.IsAccessibleToAllProjects = convertIntegrationScopeToBool(d.Get("scope").(string))
+	vars.Params.Jira = &wiz.CreateJiraIntegrationParamsInput{}
+	vars.Params.Jira.ServerURL = d.Get("jira_url").(string)
+	vars.Params.Jira.ServerType = d.Get("jira_server_type").(string)
+	vars.Params.Jira.IsOnPrem = d.Get("jira_is_on_prem").(bool)
+	vars.Params.Jira.TLSConfig.AllowInsecureTLS = utils.ConvertBoolToPointer(d.Get("jira_allow_insecure_tls").(bool))
+	vars.Params.Jira.TLSConfig.ClientCertificateAndPrivateKey = d.Get("jira_client_certificate_and_private_key").(string)
+	vars.Params.Jira.TLSConfig.ServerCA = d.Get("jira_server_ca").(string)
+	vars.Params.Jira.Authorization.Username = d.Get("jira_username").(string)
+	vars.Params.Jira.Authorization.Password = d.Get("jira_password").(string)
+
+	// process the request
+	data := &CreateIntegration{}
+	requestDiags := client.ProcessRequest(ctx, m, vars, data, query, "integration_jira", "create")
+	diags = append(diags, requestDiags...)
+	if len(diags) > 0 {
+		return diags
+	}
+
+	// set the id
+	d.SetId(data.CreateIntegration.Integration.ID)
+
+	return resourceWizIntegrationJiraRead(ctx, d, m)
+}
+
+func resourceWizIntegrationJiraRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	tflog.Info(ctx, "resourceWizIntegrationJiraRead called...")
+
+	// check the id
+	if d.Id() == "" {
+		return nil
+	}
+
+	// define the graphql query
+	query := `query integration (
+	  $id: ID!
+	) {
+	  integration(
+	    id: $id
+	  ) {
+	    id
+	    name
+	    createdAt
+	    updatedAt
+	    project {
+	      id
+	    }
+	    type
+	    isAccessibleToAllProjects
+	    usedByRules {
+	      id
+	    }
+	    paramsType: params {
+	      type: __typename
+	    }
+	    params {
+	      ... on JiraIntegrationParams {
+	        url
+			serverType
+			onPremConfig {
+				isOnPrem
+			}
+			tlsConfig {
+				allowInsecureTLS
+				serverCA
+				clientCertificateAndPrivateKey
+			}
+	        authorization {
+	          ... on JiraIntegrationBasicAuthorization {
```

Comment:
Are there plans to support JiraIntegrationTokenBearerAuthorization with this resource?

Reply:
Ah, good catch. 

We're using Jira cloud, which even though it uses a "token" when you manually set up an integration it's really just using basic authorization. I did add the option to use an on-prem server elsewhere but didn't really have a solid way of running acceptance tests on an on-prem config without having one so I stopped short of adding that in.

That being said I can still add that option, I just won't be able to directly test it since the cloud config only uses BasicAuth from what I can tell and I don't have an on-prem to run acceptance tests against. I'll see if I can add it regardless since I'm sure that would be useful for other folks.

Reply:
added